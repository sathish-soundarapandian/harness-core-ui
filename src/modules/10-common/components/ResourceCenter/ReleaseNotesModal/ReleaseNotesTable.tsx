/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { HTMLTable } from '@blueprintjs/core'
import {
  ModuleVersionsListResponseResponse,
  ModuleVersionsResponse,
  useListModuleVersions
} from 'services/cd-ng-open-api'
import { useStrings } from 'framework/strings'
import css from './UseReleaseNotesModal.module.scss'

export interface ModuleVersionTableProps {
  moduleVersions?: ModuleVersionsListResponseResponse
}

export const ModuleVersionTable: React.FC<ModuleVersionTableProps> = () => {
  const { getString } = useStrings()
  const { data } = useListModuleVersions(
    {
      queryParams: {
        page: 0
      }
    } || []
  )

  const modulesList = React.useMemo((): ModuleVersionsResponse[] | null => {
    if (!data) return null
    const orderedModulesMap = new Map<string, ModuleVersionsResponse>([
      ['CI', {}],
      ['CD', {}],
      ['CE', {}],
      ['FF', {}],
      ['SRM', {}],
      ['SRT', {}],
      ['CHAOS', {}],
      ['Platform', {}],
      ['Delegate', {}]
    ])
    data?.forEach((moduleData: ModuleVersionsResponse) => {
      orderedModulesMap.set(moduleData.name as string, moduleData)
    })
    return [...orderedModulesMap.values()]
  }, [data])
  return (
    <HTMLTable className={css.table} data-testid="release-note-table">
      <thead>
        <tr style={{ backgroundColor: '#E2F5FF' }} data-testid="release-note-table-head">
          <th className={css.th}>{getString('common.resourceCenter.ticketmenu.component')}</th>
          <th className={css.th}>{getString('version')}</th>
          <th className={css.th}>{getString('lastUpdated')}</th>
          <th className={css.th}>{getString('common.resourceCenter.bottomlayout.releaseNote')}</th>
        </tr>
      </thead>
      <tbody>
        {modulesList?.map((module: ModuleVersionsResponse, i: number) => (
          <tr key={`${module?.name}-${i}`} className={css.tr}>
            <td className={css.td} data-testid="name">
              {module?.display_name}
            </td>
            <td className={css.td} data-testid="version">
              {module?.version}
            </td>
            <td className={css.td} data-testid="updated">
              {module?.updated}
            </td>
            <td className={css.td} data-testid="link">
              <a href={`${module?.release_notes_link}`} target="_blank" rel="noreferrer noopener">
                {'Link'}
              </a>
            </td>
          </tr>
        ))}
      </tbody>
    </HTMLTable>
  )
}
