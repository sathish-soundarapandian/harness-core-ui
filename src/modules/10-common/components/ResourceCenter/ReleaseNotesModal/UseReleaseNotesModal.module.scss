/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.modal {
  .alignCenter {
    align-items: center;
  }
}

.dialog {
  border-radius: 4px;
  width: 916px;
  display: inline-table;
}

.table {
  width: inherit;
  padding: 20px;
}

.tr:nth-child(even) {
  background-color: var(--grey-100);
}

.th {
  border-bottom: 1px solid var(--black);
  border: 2px solid var(--black);
}

.td {
  text-align: center;
  border: 2px solid var(--black);
}
