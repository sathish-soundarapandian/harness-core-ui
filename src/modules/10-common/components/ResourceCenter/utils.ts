/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import moment from 'moment'
import type { UserInfo } from 'services/cd-ng'

export const ON_PREM_RELEASE_NODE_LINK = 'https://ngdocs.harness.io/article/556wy85kbo-harness-on-prem-release-notes'
export const SAAS_RELEASE_NODE_LINK = 'https://ngdocs.harness.io/article/7zkchy5lhj-harness-saa-s-release-notes-2022'

export const HARNESS_SEARCH_LINK = 'https://harness.io/search/'
export const HARNESS_UNIVERISITY_LINK = 'https://university.harness.io/page/job-role-certifications'
export const HARNESS_COMMUNITY_LINK = 'https://community.harness.io/'
export const HARNESS_COMMUNITY_SLACK_LINK =
  'https://join.slack.com/t/harnesscommunity/shared_invite/zt-y4hdqh7p-RVuEQyIl5Hcx4Ck8VCvzBw'
export const HARNESS_DOCS_LINK = 'https://docs.harness.io/'
export const HARNESS_API_DOCS_LINK = 'https://apidocs.harness.io/'
export const HARNESS_TUTORIALS = 'https://developer.harness.io/tutorials/get-started'
export const SITE_STATUS_LINK = 'https://status.harness.io/'

export const timestamp = moment.now()
export const HARNESS_SUPPORT_LINK =
  '/sso.html?action=login&brand_id=114095000394&locale_id=1&return_to=https%3A%2F%2Fsupport.harness.io%2Fhc%2Fen-us%2Frequests&src=zendesk&timestamp=' +
  timestamp

export const WHATS_NEW_LINK = `https://docs.harness.io/article/ueeiji09do-what-s-new`
export const EARLY_ACCESS_LINK = `https://docs.harness.io/article/w4krvu96i3-early-access`

export const openZendeskSupport = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(HARNESS_SUPPORT_LINK)
}

export function getReleaseNodeLink(): string {
  switch (window.deploymentType) {
    case 'COMMUNITY':
    case 'ON_PREM': {
      return ON_PREM_RELEASE_NODE_LINK
    }
    default:
      return SAAS_RELEASE_NODE_LINK
  }
}
export const openFileATicket = (
  e: React.MouseEvent<Element, MouseEvent>,
  currentUserInfo: UserInfo,
  setShow?: React.Dispatch<React.SetStateAction<boolean>>
): void => {
  e.stopPropagation()
  e.preventDefault()
  window.Saber.do?.('set_options', {
    feedback_values: {
      Email: currentUserInfo.email // set default value for email field
    }
  })
  setShow?.(false)
  window.Saber.do?.('open')
}
export const openWhatsNew = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(WHATS_NEW_LINK)
}
export const openEarlyAccess = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(EARLY_ACCESS_LINK)
}
