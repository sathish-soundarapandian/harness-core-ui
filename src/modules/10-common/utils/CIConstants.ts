/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */
export const DEFAULT_ORG_ID = 'default'

export const DEFAULT_PROJECT_NAME = 'Default Project'

export const DEFAULT_PROJECT_ID = 'Default_Project'

export const DEFAULT_ORG_NAME = 'Default Organization'

export enum Status {
  SUCCESS = 'SUCCESS',
  ERROR = 'ERROR',
  FAILURE = 'FAILURE'
}
