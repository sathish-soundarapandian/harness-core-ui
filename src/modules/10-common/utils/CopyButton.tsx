/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import { Button, ButtonProps, ButtonVariation } from '@harness/uicore'
import { Color } from '@harness/design-system'
import copy from 'clipboard-copy'
import { useStrings } from 'framework/strings'

interface CopyButtonProps extends ButtonProps {
  textToCopy: string
  onCopySuccess?: () => void
  primaryBtn?: boolean
}

const CopyButton: React.FC<CopyButtonProps> = ({ textToCopy, onCopySuccess, ...rest }) => {
  const { getString } = useStrings()
  const [openTooltip, setOpenTooltip] = useState(false)
  const showCopySuccess = () => {
    setOpenTooltip(true)
    setTimeout(
      /* istanbul ignore next */ () => {
        setOpenTooltip(false)
      },
      1000
    )
  }

  return (
    <Button
      variation={rest.primaryBtn ? ButtonVariation.PRIMARY : undefined}
      minimal
      iconProps={{ color: rest.primaryBtn ? Color.WHITE : undefined }}
      icon="duplicate"
      onClick={() => {
        copy(textToCopy)
        showCopySuccess()
        if (onCopySuccess) {
          onCopySuccess()
        }
      }}
      tooltip={getString('common.copied')}
      tooltipProps={{ isOpen: openTooltip, isDark: true }}
      {...rest}
    />
  )
}

export default CopyButton
