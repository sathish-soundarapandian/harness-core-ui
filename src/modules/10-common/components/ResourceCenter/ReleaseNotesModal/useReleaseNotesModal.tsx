/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Dialog, Divider } from '@blueprintjs/core'
import { Layout, Text } from '@harness/uicore'
import { useModalHook } from '@harness/use-modal'
// eslint-disable-next-line no-restricted-imports
import { useParams } from 'react-router'
import { String } from 'framework/strings'
import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
import { useGetAccountNG } from 'services/cd-ng'
import { ModuleVersionTable } from './ReleaseNotesTable'
import css from './UseReleaseNotesModal.module.scss'

interface ModalReturn {
  showModal: () => void
  hideModal: () => void
}

export const useReleaseNotesModal = (): ModalReturn => {
  const { accountId } = useParams<AccountPathProps>()
  const { data } = useGetAccountNG({
    accountIdentifier: accountId,
    queryParams: { accountIdentifier: accountId }
  })

  const clusterName = data?.data?.cluster

  const [showModal, hideModal] = useModalHook(() => {
    return (
      <Dialog
        isOpen
        enforceFocus={false}
        title={
          <Layout.Vertical spacing="small" className={css.alignCenter}>
            <Text font={{ size: 'medium' }}>
              <String stringID="common.resourceCenter.bottomlayout.releaseNote" />
            </Text>
            <Divider />
            <Text font={{ size: 'medium' }}>
              <String
                stringID="common.resourceCenter.productUpdates.releaseText"
                style={{ fontStyle: 'italic' }}
                useRichText
                vars={{
                  accountid: accountId,
                  clustername: clusterName
                }}
              />
            </Text>
          </Layout.Vertical>
        }
        onClose={hideModal}
        className={css.dialog}
      >
        <ModuleVersionTable />
      </Dialog>
    )
  }, [])

  return {
    showModal,
    hideModal
  }
}
