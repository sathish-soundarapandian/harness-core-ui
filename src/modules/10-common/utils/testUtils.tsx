/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { act, fireEvent, getByText, queryByAttribute, waitFor } from '@testing-library/react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation, useHistory } from 'react-router-dom'
import { ModalProvider } from '@harness/use-modal'
import qs from 'qs'
import { noop } from 'lodash-es'
import { enableMapSet } from 'immer'
import userEvent from '@testing-library/user-event'
import { AppStoreContext, AppStoreContextProps } from 'framework/AppStore/AppStoreContext'
import { LicenseStoreContext, LicenseStoreContextProps } from 'framework/LicenseStore/LicenseStoreContext'
import { LICENSE_STATE_VALUES } from 'framework/LicenseStore/licenseStoreUtil'
import { withAccountId, accountPathProps } from '@common/utils/routeUtils'
import type { Project } from 'services/cd-ng'
import { StringsContext, UseStringsReturn } from 'framework/strings'

import { FeaturesContext, FeaturesContextProps } from 'framework/featureStore/FeaturesContext'
import type { FeatureDetail, FeatureMetaData } from 'framework/featureStore/featureStoreUtil'
import type { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
import './testUtils.scss'
import { PermissionsContext, PermissionsContextProps } from 'framework/rbac/PermissionsContext'
import { Editions } from '@common/constants/SubscriptionTypes'
import type { FeatureFlag } from '@common/featureFlags'
import { PreferenceStoreContext } from 'framework/PreferenceStore/PreferenceStoreContext'
import {
  TemplateSelectorContext,
  TemplatesSelectorContextInterface
} from 'framework/Templates/TemplateSelectorContext/TemplateSelectorContext'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseGetMockDataWithMutateAndRefetch<T> extends UseGetMockData<T> {
  mutate: () => Record<string, unknown>
  refetch: () => Record<string, unknown>
  cancel?: () => Record<string, unknown>
}

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): HTMLElement | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): HTMLElement | null => document.querySelector('.bp3-popover-content')

export interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, unknown>
  defaultAppStoreValues?: Partial<AppStoreContextProps>
  defaultLicenseStoreValues?: Partial<LicenseStoreContextProps>
  defaultPermissionValues?: Partial<PermissionsContextProps>
  defaultTemplateSelectorValues?: Partial<TemplatesSelectorContextInterface>
  defaultFeaturesValues?: Partial<FeaturesContextProps>
  defaultFeatureFlagValues?: Partial<Record<FeatureFlag, boolean>>
  projects?: Project[]
  enableBrowserView?: boolean
  stringsData?: Record<string, string>
  getString?: UseStringsReturn['getString']
}

export const prependAccountPath = (path: string): string => withAccountId(() => path)(accountPathProps)

export const CurrentLocation = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export interface BrowserViewProps {
  enable?: boolean
  children: React.ReactNode
}

export function BrowserView(props: BrowserViewProps): React.ReactElement {
  const { enable, children } = props
  const location = useLocation()
  const history = useHistory()

  if (!enable) {
    return <>{children}</>
  }

  function handlePathChange(e: React.ChangeEvent<HTMLInputElement>): void {
    history.replace(e.currentTarget.value)
  }

  const search = location.search ? `?${location.search.replace(/^\?/, '')}` : ''

  return (
    <div className="browser">
      <div className="browser-header">
        <input className="browser-path" value={location.pathname + search} onChange={handlePathChange} />
      </div>
      <div className="browser-content">{children}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  enableMapSet()
  const {
    path = '/',
    pathParams = {},
    defaultAppStoreValues,
    queryParams = {},
    defaultLicenseStoreValues,
    defaultPermissionValues,
    defaultTemplateSelectorValues,
    defaultFeaturesValues,
    defaultFeatureFlagValues = {},
    stringsData = {},
    getString = (key: string) => key
  } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const history = React.useMemo(() => createMemoryHistory({ initialEntries: [routePath] }), [])

  const defaultReturn = {
    enabled: true
  }

  /** TODO: Try fixing this later. This is causing some tests to fail */
  // React.useEffect(() => {
  //   history.replace(compile(path)(pathParams) + qs.stringify(queryParams, { addQueryPrefix: true }))
  //   // eslint-disable-next-line react-hooks/exhaustive-deps
  // }, [path, pathParams, queryParams])

  return (
    <Router history={history}>
      <StringsContext.Provider value={{ data: stringsData as any, getString }}>
        <PreferenceStoreContext.Provider
          value={{
            set: noop,
            get: noop,
            clear: noop
          }}
        >
          <AppStoreContext.Provider
            value={{
              featureFlags: {
                FEATURE_ENFORCEMENT_ENABLED: true,
                ...defaultFeatureFlagValues
              },
              updateAppStore: () => void 0,
              currentUserInfo: { uuid: '' },
              ...defaultAppStoreValues
            }}
          >
            <LicenseStoreContext.Provider
              value={{
                versionMap: {},
                licenseInformation: {},
                CI_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
                FF_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
                CCM_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
                CD_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
                CHAOS_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
                STO_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
                CV_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
                CET_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
                updateLicenseStore: () => void 0,
                ...defaultLicenseStoreValues
              }}
            >
              <PermissionsContext.Provider
                value={{
                  permissions: new Map<string, boolean>(),
                  requestPermission: () => void 0,
                  checkPermission: () => true,
                  cancelRequest: () => void 0,
                  ...defaultPermissionValues
                }}
              >
                <FeaturesContext.Provider
                  value={{
                    features: new Map<FeatureIdentifier, FeatureDetail>(),
                    featureMap: new Map<FeatureIdentifier, FeatureMetaData>(),
                    getEdition: () => {
                      return undefined
                    },
                    requestFeatures: () => void 0,
                    requestLimitFeature: () => void 0,
                    checkFeature: () => {
                      return defaultReturn
                    },
                    checkLimitFeature: () => {
                      return defaultReturn
                    },
                    getRestrictionType: () => {
                      return undefined
                    },
                    getHighestEdition: () => {
                      return Editions.FREE
                    },
                    ...defaultFeaturesValues
                  }}
                >
                  <ModalProvider>
                    <TemplateSelectorContext.Provider
                      value={{
                        state: {
                          isDrawerOpened: false,
                          selectorData: {
                            templateType: 'Step',
                            onSubmit: jest.fn(),
                            onCancel: jest.fn()
                          }
                        },
                        openTemplateSelector: jest.fn(),
                        closeTemplateSelector: jest.fn(),
                        ...defaultTemplateSelectorValues
                      }}
                    >
                      <RestfulProvider base="/">
                        <BrowserView enable={props.enableBrowserView}>
                          <Switch>
                            <Route exact path={path}>
                              {props.children}
                            </Route>
                            <Route>
                              <CurrentLocation />
                            </Route>
                          </Switch>
                        </BrowserView>
                      </RestfulProvider>
                    </TemplateSelectorContext.Provider>
                  </ModalProvider>
                </FeaturesContext.Provider>
              </PermissionsContext.Provider>
            </LicenseStoreContext.Provider>
          </AppStoreContext.Provider>
        </PreferenceStoreContext.Provider>
      </StringsContext.Provider>
    </Router>
  )
}

export const queryByNameAttribute = (name: string, container: HTMLElement): HTMLElement | null =>
  queryByAttribute('name', container, name)

export const doConfigureOptionsTesting = async (
  cogModal: HTMLElement,
  fieldElement: HTMLInputElement
): Promise<void> => {
  // Type regex and submit
  // check if field has desired value
  await waitFor(() => expect(getByText(cogModal, 'common.configureOptions.regex')).toBeInTheDocument())
  const regexRadio = getByText(cogModal, 'common.configureOptions.regex')
  userEvent.click(regexRadio)
  const regexTextArea = queryByAttribute('name', cogModal, 'regExValues') as HTMLInputElement
  act(() => {
    fireEvent.change(regexTextArea, { target: { value: '<+input>.includes(/test/)' } })
  })
  const cogSubmit = getByText(cogModal, 'submit')
  userEvent.click(cogSubmit)
  await waitFor(() => expect(fieldElement.value).toBe('<+input>.regex(<+input>.includes(/test/))'))
}
