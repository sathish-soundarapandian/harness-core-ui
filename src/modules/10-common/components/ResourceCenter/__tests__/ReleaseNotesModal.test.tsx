/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import { TestWrapper, findDialogContainer } from '@common/utils/testUtils'
import { useGetAccountNG } from 'services/cd-ng'
import * as cdNgOpenApiServices from 'services/cd-ng-open-api'
import { useReleaseNotesModal } from '../ReleaseNotesModal/useReleaseNotesModal'

jest.mock('services/cd-ng')
const useGetAccountNGMock = useGetAccountNG as jest.MockedFunction<any>

beforeEach(() => {
  window.deploymentType = 'SAAS'

  useGetAccountNGMock.mockImplementation(() => {
    return {
      data: {
        data: {
          name: 'account name',
          identifier: 'id1',
          cluster: 'free',
          defaultExperience: 'NG'
        }
      },
      refetch: jest.fn()
    }
  })

  jest.spyOn(cdNgOpenApiServices, 'useListModuleVersions').mockReturnValue({
    data: [
      {
        name: 'testName',
        display_name: 'testDisplayName',
        version: '7800',
        updated: 'testUpdate',
        release_notes_link: 'link'
      }
    ],
    refetch: jest.fn(),
    loading: false,
    error: null
  } as any)
})

const TestComponent = (): React.ReactElement => {
  const { showModal, hideModal } = useReleaseNotesModal()
  return (
    <>
      <button className="open" onClick={showModal} />
      <button className="close" onClick={hideModal} />
    </>
  )
}

describe('open and close release notes  Modal', () => {
  describe('Rendering', () => {
    test('should open and close the release notes modal', async () => {
      const { container, getByText } = render(
        <TestWrapper>
          <TestComponent />
        </TestWrapper>
      )
      fireEvent.click(container.querySelector('.open')!)

      const dialog = findDialogContainer() as HTMLElement
      expect(dialog).toMatchSnapshot()

      await waitFor(() => {
        expect(getByText('common.resourceCenter.bottomlayout.releaseNote')).toBeInTheDocument()
        expect(getByText('common.resourceCenter.productUpdates.releaseText')).toBeInTheDocument()
      })
    })
  })
})
