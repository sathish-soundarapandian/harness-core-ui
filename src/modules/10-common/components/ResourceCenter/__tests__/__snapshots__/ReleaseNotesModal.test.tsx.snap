// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open and close release notes  Modal Rendering should open and close the release notes modal 1`] = `
<div
  class="bp3-dialog dialog"
>
  <div
    class="bp3-dialog-header"
  >
    <h4
      class="bp3-heading"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-small StyledProps--main alignCenter"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium"
        >
          <span>
            common.resourceCenter.bottomlayout.releaseNote
          </span>
        </p>
        <div
          class="bp3-divider"
        />
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium"
        >
          <span
            style="font-style: italic;"
          >
            common.resourceCenter.productUpdates.releaseText
          </span>
        </p>
      </div>
    </h4>
    <button
      aria-label="Close"
      class="bp3-button bp3-minimal bp3-dialog-close-button"
      type="button"
    >
      <span
        class="bp3-icon bp3-icon-small-cross"
        icon="small-cross"
      >
        <svg
          data-icon="small-cross"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <desc>
            small-cross
          </desc>
          <path
            d="M11.41 10l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L10 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L8.59 10 5.3 13.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l3.29-3.3 3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L11.41 10z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
  <table
    class="bp3-html-table table"
    data-testid="release-note-table"
  >
    <thead>
      <tr
        data-testid="release-note-table-head"
        style="background-color: rgb(226, 245, 255);"
      >
        <th
          class="th"
        >
          common.resourceCenter.ticketmenu.component
        </th>
        <th
          class="th"
        >
          version
        </th>
        <th
          class="th"
        >
          lastUpdated
        </th>
        <th
          class="th"
        >
          common.resourceCenter.bottomlayout.releaseNote
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        />
        <td
          class="td"
          data-testid="version"
        />
        <td
          class="td"
          data-testid="updated"
        />
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="undefined"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
          data-testid="name"
        >
          testDisplayName
        </td>
        <td
          class="td"
          data-testid="version"
        >
          7800
        </td>
        <td
          class="td"
          data-testid="updated"
        >
          testUpdate
        </td>
        <td
          class="td"
          data-testid="link"
        >
          <a
            href="link"
            rel="noreferrer noopener"
            target="_blank"
          >
            Link
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
